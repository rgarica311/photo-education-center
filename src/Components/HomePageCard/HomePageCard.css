.home-page-card {
  background-image: url('./transparentDiv.png');
  width: 326px;
  height: 132px;
  display: flex;
  background-size: cover;
  background-repeat: repeat;
}

@media screen and (max-width: 600px) {
  .home-page-card {
    width: 310px;
    margin-top: 30px;
  }
}
