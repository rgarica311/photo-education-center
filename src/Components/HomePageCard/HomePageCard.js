import React from 'react';
import { Link } from 'react-router-dom';

import transparentDiv from './transparentDiv.png';
import './HomePageCard.css'



const spanStyle = {
  fontFamily: 'Kadwa, serif',
  color: 'white',
  fontSize: '1.4rem',
  lineHeight: '1.2em',
  marginBottom: '15px',
  textTransform: 'uppercase',
};

const content = {
  display: 'flex',
  flexDirection: 'column',
  marginLeft: '20px',
  marginTop: '30px',
  position: 'relative',
};

const viewMoreContainer = {
  postion: 'absolute',
  bottom: '3px'
}

const svgStyle = {
  color: 'red',
  marginTop: '30px',
  marginLeft: '20px'
}

const linkStyle = {
  textDecoration: 'none',
  color: 'red',
  position: 'absolute',
  bottom: '15px'
}

export default function HomePageCard( props ) {
  console.log( 'home page card props', props.feature.icon );
  return (
    <>
      <div className='home-page-card'>
        <svg style={ svgStyle }
            aria-hidden="true"
            focusable="false"
            data-prefix="fas"
            data-icon="camera"
            className="svg-inline--fa fa-camera fa-3x"
            role="img"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 512 512">
            <path
            fill="currentColor"
            d="M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z">
            </path>
        </svg>
        <div style={content} className='contentFeature'>
          <span style={spanStyle}>{props.feature.className}</span>
          <Link style={linkStyle} to={props.feature.path}>View More</Link>
        </div>
      </div>

    </>
  );
}
