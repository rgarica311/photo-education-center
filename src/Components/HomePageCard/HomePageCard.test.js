import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HomePageCard from './HomePageCard';

describe(`HomePageCard Component`, () => {
  const feature = {
    className: 'test class name',
    path: '/testpath'
  }
  it('renders HomePageCard by default', () => {
    const wrapper = shallow(<HomePageCard feature={feature}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
