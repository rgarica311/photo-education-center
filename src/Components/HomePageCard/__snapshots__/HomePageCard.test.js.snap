// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageCard Component renders HomePageCard by default 1`] = `
<Fragment>
  <div
    className="home-page-card"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-camera fa-3x"
      data-icon="camera"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={
        Object {
          "color": "red",
          "marginLeft": "20px",
          "marginTop": "30px",
        }
      }
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z"
        fill="currentColor"
      />
    </svg>
    <div
      className="contentFeature"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "marginLeft": "20px",
          "marginTop": "30px",
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "color": "white",
            "fontFamily": "Kadwa, serif",
            "fontSize": "1.4rem",
            "lineHeight": "1.2em",
            "marginBottom": "15px",
            "textTransform": "uppercase",
          }
        }
      >
        test class name
      </span>
      <Link
        style={
          Object {
            "bottom": "15px",
            "color": "red",
            "position": "absolute",
            "textDecoration": "none",
          }
        }
        to="/testpath"
      >
        View More
      </Link>
    </div>
  </div>
</Fragment>
`;
